src/corelibs/U2View/src/ov_msa/MSAEditor.h

```diff
@@ -102,6 +114,17 @@ class U2VIEW_EXPORT MSAEditor : public MaEditor {
 
     void sortSequences(const MultipleAlignment::SortType &sortType, const MultipleAlignment::Order &sortOrder);
 
+    /** Forces complete re-computation of the active collapse model based on the current MSA editor state. */
+    void updateCollapseModel();
+
+    void setRowOrderMode(MaEditorRowOrderMode mode) override;
+
+    /**
+     * Returns modifiable instance of the 'freeModeMasterMarkersSet'.
+     * TODO: change signature to return const reference.
```

Comment:
Where did you get this `TODO` from?

Reply:
I added it after reading existing code that works with freeModeMasterMarkerSet - this set should be protected in some way and not public.
I fixed the issue now and removed the TODO.